### PR TITLE
Updated to replace GS Collections with Eclipse Collections

### PR DESCRIPTION
**Pull request by mlalma**

### CODE REVIEW

**Comment by jvirtanen** [comment]
Looking good! Just one small thing. Currently all dependencies are defined in the top-level POM, mainly in order to keep the dependency versions in different Maven modules consistent. So, it would be good if

1. Eclipse Collections would be defined in the top-level POM, and
2. the dependency version would be defined as a property (see `jmh.version` for an example).

**Comment by jvirtanen** [comment]
(The dependency version should be defined as a property, because it is used for two dependencies.)

**Comment by mlalma** [comment]
Added Eclipse Framework dependencies to top-level DOM, added Eclipse Framework version to properties structure on parity top-level pom.xml (eclfw.version).

Removed Eclipse Framework version from parity-client's pom.xml.

File: pom.xml [hunk 1]
```diff
@@ -36,6 +36,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <java.version>1.8</java.version>
     <jmh.version>0.5.6</jmh.version>
+    <eclfw.version>7.0.0</eclfw.version>
```
Comment (jvirtanen) [inline pom.xml]: What does this stand for? Perhaps this could be also `eclipse.collections.version` for clarity.

Comment (jvirtanen) [inline pom.xml]: Eclipse has many frameworks but I guess this version number is unique to Eclipse Collections, right?

**Comment by mlalma** [comment]
Good point; I don't know what I was thinking, when I renamed Eclipse Collections to Eclipse Framework. Fixed it.

**Comment by jvirtanen** [comment]
Almost there! I noticed there are some tabs still mixed with spaces in the indentation; see the inline comments for details.

File: pom.xml [hunk 2]
```diff
@@ -54,9 +55,14 @@
   <dependencyManagement>
     <dependencies>
       <dependency>
-        <groupId>com.goldmansachs</groupId>
-        <artifactId>gs-collections</artifactId>
-        <version>5.1.0</version>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections-api</artifactId>
+    	<version>${eclipse.collections.version}</version>
```
Comment (jvirtanen) [inline pom.xml]: Tabs used in this block.

File: pom.xml [hunk 3]
```diff
@@ -54,9 +55,14 @@
...
+      </dependency>
+      <dependency>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections</artifactId>
+    	<version>${eclipse.collections.version}</version>
```
Comment (jvirtanen) [inline pom.xml]: Tabs used in this block.

File: parity-client/pom.xml [hunk 4]
```diff
@@ -53,6 +49,14 @@
       <groupId>org.jvirtanen.nassau</groupId>
       <artifactId>nassau</artifactId>
     </dependency>
+    <dependency>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections-api</artifactId>
```
Comment (jvirtanen) [inline parity-client/pom.xml]: Tabs used in this block.

File: parity-client/pom.xml [hunk 5]
```diff
@@ -53,6 +49,14 @@
...
+    </dependency>
+    <dependency>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections</artifactId>
```
Comment (jvirtanen) [inline parity-client/pom.xml]: Tabs used in this block.

**Comment by jvirtanen** [comment]
A tip: `git diff` highlights mixed spaces and tabs prominently in its output.

**Comment by mlalma** [comment]
Replaced tabs with spaces on pom.xml files

**Comment by jvirtanen** [comment]
Merged, thank you!

**Pull request merged**